### PR TITLE
297/1248: Prisoner Capture Consolidation

### LINKED ISSUE #297: Still capturing prisoners even though "random enemy capture" is set to off
Even though i have random enemy capture set to off, i am still capturing prisoners for some reason? See the campaign save and daily log.

[test.gz](https://github.com/MegaMek/mekhq/files/679696/test.gz)

### LINKED ISSUE #1248: Near 100% Defection Rate
### Environment
0.46.0-rc.2
Java build 1.8.0_221-b11
Win10

### Description
Nearly all captured prisoners are willing to defect.

### Files
Pre-battle campaign save:
[Steel Recon30291020.zip](https://github.com/MegaMek/mekhq/files/3511527/Steel.Recon30291020.zip)

Last turn of the battle autosave :
[autosave.sav.gz](https://github.com/MegaMek/mekhq/files/3511529/autosave.sav.gz)

[customs.zip](https://github.com/MegaMek/mekhq/files/3511528/customs.zip)

Prisoners

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #297 and fixes #1248 by implementing a prisoner capture status enum. I also used the time to enable further customization and modularization of the AtB Prisoner System, so that you can largely use it without playing AtB.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/enums/PrisonerCaptureStyle.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    //region Boolean Comparisons
+    public boolean isEnabled() {
+        return this != NONE;
+    }
+
+    public boolean isAtB() {
+        return this == ATB;
+    }
+
+    public boolean isTaharqa() {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/enums/PrisonerCaptureStyle.java:55]: Where's this used, anyway?

**Review by NickAragua — COMMENTED** [review]
Probably fine, I'm curious about the inclusion of the unused "Taharqa style prisoner capture" setting.

File: MekHQ/src/mekhq/campaign/personnel/enums/PrisonerCaptureStyle.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/enums/PrisonerCaptureStyle.java]: Forgot to remove it

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 2]
```diff
@@ -6231,6 +6231,17 @@ public int getUnitRatingMod() {
                         : rating.getModifier();
     }
 
+    /**
+     * This is a better method for pairing AtB with IOpts with regards to Prisoner Capture
+     */
+    public int getUnitRatingAsInteger() {
+        if (getCampaignOptions().useDragoonRating()) {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/Campaign.java:6238]: This seems like if you're using the CamOps unit rating, then it'll always return a rating C, which may not be the desired effect.

**Review by NickAragua — COMMENTED** [review]
So if I'm reading this right, there's a "Taharqa" prisoner capture style and an "AtB" prisoner capture style. However, I don't really see any difference between those two prisoner capture styles in practical terms. Is there something I'm missing, or is it kind of redundant?

File: MekHQ/src/mekhq/campaign/Campaign.java (hunk 2 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/Campaign.java:6238]: Confusingly named variable... useDragoonsRating enables Unit Rating. I've got an idea for a rework of it... but not here nor now.

**Comment by Windchild292** [comment]
Taharqa is you capture everyone possible from certain restrictions. AtB has any possible captures (which are thankfully determined by Taharqa's method) rolled, and only those that roll successfully show up. Bloody confusing setup, but... it works.

**Comment by Windchild292** [comment]
Updated the Taharqa method tooltip to make it easier to understand the difference.

**Review by sixlettervariables — APPROVED** [review]

**Pull request merged**